pallets/parachain-staking/src/lib.rs

```diff
@@ -571,7 +579,9 @@ pub mod pallet {
 		/// The first block of the current round
 		pub first: BlockNumber,
 		/// The length of the current round in number of blocks
-		pub length: u32,
+		pub length: BlockNumber,
```

Comment:
This is the cursed change. It seems so straight-forward and natural. But it caused so many issues.